modules/images/webp-uploads/image-edit.php

```diff
@@ -129,16 +166,38 @@ function ( $metadata, $post_meta_id ) use ( $post_id, $file_path, $mime_type, $e
 					continue;
 				}
 
-				$extension   = explode( '|', $allowed_mimes[ $targeted_mime ] );
-				$destination = trailingslashit( $original_directory ) . "{$filename}.{$extension[0]}";
-				$result      = $editor->save( $destination, $targeted_mime );
-
-				if ( is_wp_error( $result ) ) {
-					continue;
+				$extension = explode( '|', $allowed_mimes[ $targeted_mime ] );
+				$extension = $extension[0];
+
+				if ( 'thumbnail' === $target ) {
```

Comment:
Let's add a similar comment here to explain:
```suggestion
				// If the target is `thumbnail` make sure only that size is generated.
				if ( 'thumbnail' === $target ) {
```